worker_test.go

```diff
@@ -122,54 +146,100 @@ func TestExitCode(t *testing.T) {
 }
 
 func TestResultCache(t *testing.T) {
-	result1, err := postJob(JobRequest{"result-cache", params, ""})
+	target, err := ioutil.TempDir("", "mistry-tests-results")
+	if err != nil {
+		log.Fatal(err)
+	}
+	user, err := user.Current()
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
This is needed in various tests, so you could do this once in `TestMain()`

Reply:
True, extracted in `TestMain()` and `init()` as appropriate.